ibazel/profiler/profiler.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){
+  function extractUrl() {
+    const reducer = function(
+        accumulator, currentValue) { return accumulator || currentValue; };
+    return [...window.document.getElementsByTagName('script') ]
+        .map(function(element) {
+          if (element.src) {
```

Comment:
You're mixing filtering and mapping logic in here. You could also do

```
    return [...window.document.getElementsByTagName('script') ]
        .filter(function(e) { return element.src})
￼        .map(function(e) {
           const parsed = document.createElement('a');
           parsed.href = e.src;
           return parsed;
         })
        .filter(function(e) { return e.src})
        .map(function(e) {
           return `${parsed.protocol}//${parsed.host}/profiler-event`;
         })
        .reduce(function(acc, u) {...})
```

and then not have that if statement

Reply:
Much cleaner. Thanks.